pkg/ingestor/parser/vuln/vuln.go

```diff
@@ -0,0 +1,179 @@
+// exists.
+//
+// An attestation node is generated that contains the information stored in the
+// attestation document.  The attestation node is linked to the package node
+// via a "attestation" edge type.
+//
+// Depending on the number of vulnerabilities found, vulnerability nodes are
+// generated that contain just the vulnerability ID that can be used to query
+// for more information as needed.  The vulnerability node is linked to the
+// attestation node via a "vulnerable" edge type.
+package vuln
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/guacsec/guac/pkg/assembler"
+	"github.com/guacsec/guac/pkg/assembler/clients/generated"
+	"github.com/guacsec/guac/pkg/assembler/helpers"
+	attestation_vuln "github.com/guacsec/guac/pkg/certifier/attestation"
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/guacsec/guac/pkg/ingestor/parser/common"
+)
+
+type parser struct {
+	packages  []*generated.PkgInputSpec
+	vulnData  *generated.VulnerabilityMetaDataInput
+	vulns     []*generated.OSVInputSpec
+	isVulns   []assembler.IsVulnIngest
+	isOccs    []assembler.IsOccurenceIngest
+	artifacts []*generated.ArtifactInputSpec
+}
+
+// NewVulnCertificationParser initializes the parser
+func NewVulnCertificationParser() common.DocumentParser {
+	return &parser{}
+}
+
+// Parse breaks out the document into the graph components
+func (c *parser) Parse(ctx context.Context, doc *processor.Document) error {
+	statement, err := parseVulnCertifyPredicate(doc.Blob)
+	if err != nil {
+		return fmt.Errorf("failed to parse slsa predicate: %w", err)
+	}
+	ps, ios, err := parseSubject(statement)
+	if err != nil {
+		return fmt.Errorf("unable to parse subject of statement: %w", err)
+	}
+	c.packages = ps
+	c.isOccs = ios
+	c.vulnData = parseMetadata(statement)
+	vs, ivs, err := parseVulns(statement)
+	if err != nil {
+		return fmt.Errorf("unable to parse vulns of statement: %w", err)
+	}
+	c.vulns = vs
+	c.isVulns = ivs
+	return nil
+}
+
+func parseVulnCertifyPredicate(p []byte) (*attestation_vuln.VulnerabilityStatement,
+	error) {
+	predicate := attestation_vuln.VulnerabilityStatement{}
+	if err := json.Unmarshal(p, &predicate); err != nil {
+		return nil, err
+	}
+	return &predicate, nil
+}
+
+func parseSubject(s *attestation_vuln.VulnerabilityStatement) ([]*generated.PkgInputSpec,
+	[]assembler.IsOccurenceIngest, error) {
+	var ps []*generated.PkgInputSpec
+	var ios []assembler.IsOccurenceIngest
+	for _, sub := range s.StatementHeader.Subject {
+		p, err := helpers.PurlToPkg(sub.Name)
+		if err != nil {
+			return nil, nil, fmt.Errorf("bad purl in statement header: %w", err)
+		}
+		ps = append(ps, p)
+		for a, d := range sub.Digest {
+			io := assembler.IsOccurenceIngest{
+				Pkg: p,
+				Artifact: &generated.ArtifactInputSpec{
+					Algorithm: a,
+					Digest:    d,
+				},
+				IsOccurence: &generated.IsOccurrenceInputSpec{
+					Justification: "Reported in vulnerability certifier", // FIXME is this ok?
```

Comment:
What should we put here?

Reply:
probably something [similar to the SLSA ](https://github.com/guacsec/guac/blob/main/pkg/ingestor/parser/slsa/parser_slsa.go#L161)one will do, more so how it was derived so folks can determine if they trust the fidelity of the field (on top of the trust info of the document)